python/pacemaker/_cts/tests/cibsecret.py

```diff
@@ -0,0 +1,176 @@
+from pacemaker._cts.timer import Timer
+
+# Disable various pylint warnings that occur in so many places throughout this
+# file it's easiest to just take care of them globally.  This does introduce the
+# possibility that we'll miss some other cause of the same warning, but we'll
+# just have to be careful.
+
+# pylint doesn't understand that self._env is subscriptable.
+# pylint: disable=unsubscriptable-object
+
+
+class CibsecretTest(CTSTest):
+    """Test managing secrets with cibsecret."""
+
+    def __init__(self, cm):
+        """
+        Create a new CibsecretTest instance.
+
+        Arguments:
+        cm -- A ClusterManager instance
+        """
+        CTSTest.__init__(self, cm)
+        self.name = "Cibsecret"
+
+        self._secret = "passwd"
+        self._secret_val = "SecreT_PASS"
+
+        self._rid = "secretDummy"
+        self._start = StartTest(cm)
+        self._startall = SimulStartLite(cm)
+
+    def _insert_dummy(self, node):
+        """Create a dummy resource on the given node."""
+        pats = [
+            f"{node}.*" + (self.templates["Pat:RscOpOK"] % ("start", self._rid))
+        ]
+
+        watch = self.create_watch(pats, 60)
+        watch.set_watch()
+
+        self._cm.add_dummy_rsc(node, self._rid)
+
+        with Timer(self._logger, self.name, "addDummy"):
+            watch.look_for_all()
+
+        if watch.unmatched:
+            self.debug("Failed to find patterns when adding dummy resource")
+            return repr(watch.unmatched)
+
+        return ""
+
+    def _check_cib_value(self, node, expected):
+        (rc, lines) = self._rsh(node, f"crm_resource -r {self._rid} -g {self._secret}",
+                                verbose=1)
+        s = " ".join(lines).strip()
+
+        if rc != 0 or s != expected:
+            return self.failure(f"Secret set to '{s}', not '{expected}'")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_check(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret check {self._rid} {self._secret}",
+                            verbose=1)
+        if rc != 0:
+            return self.failure("Failed to check secret")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_delete(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret delete {self._rid} {self._secret}",
+                            verbose=2)
+        if rc != 0:
+            return self.failure("Failed to delete secret")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_get(self, node, expected):
+        (rc, lines) = self._rsh(node, f"cibsecret get {self._rid} {self._secret}",
+                                verbose=1)
+        s = " ".join(lines).strip()
+
+        if rc != 0 or s != expected:
+            return self.failure(f"Secret set to '{s}', not '{expected}'")
```

Comment:
Maybe differentiate this from the error in `_check_cib_value()`: `"Secret set to '{s}' in local file..."` vs. `"...in CIB..."`, or something